packages/ui/src/components/Channel/index.tsx

```diff
@@ -0,0 +1,59 @@
+import * as client from '@tloncorp/shared/dist/client';
+import { KeyboardAvoidingView, Platform } from 'react-native';
+import { YStack } from 'tamagui';
+
+import { useContacts } from '../../contexts';
+import { ChannelHeader } from './ChannelHeader';
+import ChatScroll from './ChatScroll';
+import MessageInput from './MessageInput';
+
+export function Channel({
+  channel,
+  posts,
+  goBack,
+  goToChannels,
+  goToSearch,
+  // TODO: implement gallery and notebook
+  type,
+}: {
+  channel: client.Channel;
+  posts: client.Post[];
+  goBack: () => void;
+  goToChannels: () => void;
+  goToSearch: () => void;
+  type?: 'chat' | 'gallery' | 'notebook';
+}) {
+  const generateTitleFromMembers = (members: client.GroupMember[]) => {
+    const contacts = useContacts();
+    return members
+      .map((m) => contacts[m.id].nickname || contacts[m.id].id)
+      .join(', ');
+  };
+
+  const title = channel.title
```

Comment:
I'd memoize this title stuff + maybe extract as a hook or utils function -- i think we'll need it elsewhere too. Might even make sense to cache on db update.